server/src/test/java/org/elasticsearch/index/codec/PerFieldMapperCodecTests.java

```diff
@@ -86,11 +87,15 @@ public class PerFieldMapperCodecTests extends ESTestCase {
         """;
 
     public void testUseBloomFilter() throws IOException {
-        PerFieldFormatSupplier perFieldMapperCodec = createFormatSupplier(false, randomBoolean(), false);
+        boolean timeSeries = randomBoolean();
+        PerFieldFormatSupplier perFieldMapperCodec = createFormatSupplier(false, timeSeries, false);
         assertThat(perFieldMapperCodec.useBloomFilter("_id"), is(true));
         assertThat(perFieldMapperCodec.getPostingsFormatForField("_id"), instanceOf(ES87BloomFilterPostingsFormat.class));
         assertThat(perFieldMapperCodec.useBloomFilter("another_field"), is(false));
-        assertThat(perFieldMapperCodec.getPostingsFormatForField("another_field"), instanceOf(ES812PostingsFormat.class));
+        assertThat(
+            perFieldMapperCodec.getPostingsFormatForField("another_field"),
+            instanceOf(timeSeries ? ES812PostingsFormat.class : Lucene101PostingsFormat.class)
```

Comment:
I think these tests fail if build is in release mode. Can you add `test-release` label to this PR?
I think we should have an if check here. If feature flag is enabled we expect the lucene 101 posting format otherwise the es812 format.